plugins/radar-android-google-sleep/src/main/java/org.radarbase.passive.google.sleep/GoogleSleepManager.kt

```diff
@@ -0,0 +1,153 @@
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Process
+import androidx.core.content.ContextCompat
+import com.google.android.gms.location.ActivityRecognition
+import com.google.android.gms.location.SleepClassifyEvent
+import com.google.android.gms.location.SleepSegmentEvent
+import com.google.android.gms.location.SleepSegmentRequest
+import org.radarbase.android.data.DataCache
+import org.radarbase.android.source.AbstractSourceManager
+import org.radarbase.android.source.BaseSourceState
+import org.radarbase.android.source.SourceStatusListener
+import org.radarbase.android.util.SafeHandler
+import org.radarbase.android.util.toPendingIntentFlag
+import org.radarcns.kafka.ObservationKey
+import org.radarcns.passive.google.GoogleSleepClassifyEvent
+import org.radarcns.passive.google.GoogleSleepSegmentEvent
+import org.radarcns.passive.google.SleepStatus
+import org.slf4j.LoggerFactory
+
+class GoogleSleepManager(context: GoogleSleepService) : AbstractSourceManager<GoogleSleepService, BaseSourceState>(context) {
+    private val segmentEventTopic: DataCache<ObservationKey, GoogleSleepSegmentEvent> = createCache("android_google_sleep_segment_event", GoogleSleepSegmentEvent())
+    private val classifyEventTopic: DataCache<ObservationKey, GoogleSleepClassifyEvent> = createCache("android_google_sleep_classify_event", GoogleSleepClassifyEvent())
+
+    private val sleepBroadcastReceiver: BroadcastReceiver
+    private val sleepHandler = SafeHandler.getInstance("Google Sleep", Process.THREAD_PRIORITY_BACKGROUND)
+    private lateinit var sleepPendingIntent: PendingIntent
+
+    init {
+        name = context.getString(R.string.googleSleepDisplayName)
+        sleepBroadcastReceiver = SleepReceiver(this)
+    }
+
+    override fun start(acceptableIds: Set<String>) {
+        register()
+        sleepHandler.start()
+        status = SourceStatusListener.Status.READY
+        sleepHandler.execute {
+            createSleepPendingIntent()
+            registerForSleepData()
+        }
+    }
+
+     fun sendSleepClassifyData(sleepIntent: Intent) {
+        logger.info("Sleep segment event data received")
+        val sleepSegmentEvents: List<SleepSegmentEvent> = SleepSegmentEvent.extractEvents(sleepIntent)
+        sleepSegmentEvents.forEach { sleepSegmentEvent ->
+            val time = currentTime
+            val sleepStartTime: Double = sleepSegmentEvent.startTimeMillis / 1000.0
+            val sleepEndTime: Double = sleepSegmentEvent.endTimeMillis / 1000.0
+            val sleepDuration: Double = sleepSegmentEvent.segmentDurationMillis / 1000.0
+            val sleepStatus: SleepStatus = sleepSegmentEvent.status.toSleepStatus()
+
+            send(segmentEventTopic, GoogleSleepSegmentEvent(time, time, sleepStartTime,
```

Comment:
Here I would expect the first variable to be `sleepStartTime`, so
```suggestion
            send(segmentEventTopic, GoogleSleepSegmentEvent(sleepStartTime, time, sleepStartTime,
```